server/src/services/status_service.ts

```diff
@@ -19,6 +20,18 @@ export class StatusService {
         private readonly authenticationService: AuthenticationService,
     ) {}
 
+    async getStatuses(reqUser: ReqUser): Promise<Status[]> {
+        return this.getStatusesStrategy(reqUser)();
+    }
+
+    private getStatusesStrategy(reqUser: ReqUser): () => Promise<Status[]> {
+        if (reqUser.role === Role.DOCTOR) {
+            return this.statusRepository.findStatusesByDoctor.bind(this.statusRepository, reqUser.userId);
```

Comment:
what does this do?

Reply:
This allows us to partially apply the arguments to the function so that all the strategies returned have the same signature `() => Promise<Status[]>` i.e. no parameters so we can invoke them all the same.